lib/openstack_query/docs/user_docs/USERS.md

```diff
@@ -0,0 +1,68 @@
+| USER_DOMAIN_ID   | `None`  | The ID for the domain which owns the user |
+| USER_DESCRIPTION | `None`  | The description of this user.             |
+| USER_EMAIL       | `None`  | The email address of this user.           |
+| USER_ID          | `None`  | Unique ID Openstack has assigned the user |
+| USER_NAME        | `None`  | Unique user name (within the domain)      |
+
+
+Any of these properties can be used for any of the API methods that takes a property - like `select`, `where`, `sort_by` etc
+Alternatively, you can pass property aliases (passed as string) instead (currently WIP)
+
+## Chaining
+This section details valid mappings you can use to chain onto other queries or from other queries to chain into a `ProjectQuery` object.
+This applies to API calls `then` and `append_from` - see (API.md)[API.md] for details
+
+
+## Chaining from
+A `UserQuery` can be chained to other queries.
+The following shared-common properties are listed below (as well as the Query object they map to):
+
+| Prop 1                 | Prop 2                   | Type        | Maps                         |
+|------------------------|--------------------------|-------------|------------------------------|
+| UserProperties.USER_ID | ServerProperties.USER_ID | One-to-Many | `UserQuery` to `ServerQuery` |
+
+
+## Chaining to
+Chaining from other `UserQuery` requires passing `USER_QUERY` as the `query_type`
+
+| From          | Prop 1                   | Prop 2                 | Type        | Documentation            |
+|---------------|--------------------------|------------------------|-------------|--------------------------|
+| `ServerQuery` | ServerProperties.USER_ID | UserProperties.USER_ID | Many-to-One | [SERVERS.md](SERVERS.md) |
+
+
+## run() meta-parameters
+
+`UserQuery()` has the following meta-parameters that can be used when calling `run()` to fine-tune the query.
+
+| Parameter Definition                                             | Optional? | Description                                                                                                                                                                                                                                  |
+|------------------------------------------------------------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `from_domain: UserDomains` <br/><br/>(see `/enums/user_domains`) | Yes       | A User Domain to limit User query to<br/>Optional, if not given will run search on DEFAULT_DOMAIN - `UserDomains.STFC` <br/><br /> NOTE: it is possible to specify user domain via `where()` - if both are provided then an error is raised) |
+
+**NOTE: This query will only work with admin credentials - set by `clouds.yaml`**
```

Comment:
Should we add this note near the top of the page to highlight the query can only be done by admins?